crates/flux-middle/src/rty/pretty.rs

```diff
@@ -235,6 +236,32 @@ impl Pretty for SubsetTy {
     }
 }
 
+// This is a trick to avoid pretty printing `S [S { x: 10, y: 20}]`
+// and instead just print `S[{x: 10, y: 20}]` for struct-valued indices.
+struct TopIndexExpr(Expr);
+
+impl Pretty for TopIndexExpr {
+    fn fmt(&self, cx: &PrettyCx, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        define_scoped!(cx, f);
+        let e = &self.0;
+        let e = if cx.simplify_exprs { e.simplify() } else { e.clone() };
+        if let ExprKind::Aggregate(AggregateKind::Adt(def_id), flds) = e.kind()
+            && let Some(genv) = cx.genv
+            && let Ok(adt_sort_def) = genv.adt_sort_def_of(def_id)
+        {
+            let field_binds = adt_sort_def
+                .field_names()
+                .iter()
+                .zip(flds)
```

Comment:
nit: I typically prefer using `zip` as a function

```suggestion
            let field_binds = iter::zip(adt_sort_def.field_names(), flds)
```